ui/views_test.py

```diff
@@ -370,10 +403,104 @@ def test_collection_viewset_detail_as_superuser(logged_in_apiclient):
+    mock_moira_client.return_value.user_lists.return_value = [user_list_data.moira_list.name]
+    url = reverse('collection-detail', kwargs={'collection_key': user_list_data.collection.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_200_OK
+
+
+def test_collection_detail_view_permission(mock_moira_client, logged_in_apiclient, user_list_data):
+    """
+    Tests that a user cannot view a collection if user is a member of collection's view_lists
+    """
+    client, _ = logged_in_apiclient
+    user_list_data.collection.view_lists = [user_list_data.moira_list]
+    user_list_data.collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [user_list_data.moira_list.name]
+    url = reverse('collection-detail', kwargs={'collection_key': user_list_data.collection.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_403_FORBIDDEN
+
+
+def test_video_detail_view_permission(mock_moira_client, logged_in_apiclient, user_list_data):
+    """
+    Tests that a user can view a video if user is a member of collection's view_lists
+    """
+    client, _ = logged_in_apiclient
+    user_list_data.collection.view_lists = [user_list_data.moira_list]
+    user_list_data.collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [user_list_data.moira_list.name]
+    url = reverse('video-detail', kwargs={'video_key': user_list_data.video.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_200_OK
+    assert json.loads(result.context_data['js_settings_json'])['editable'] is False
+
+
+def test_video_detail_admin_permission(logged_in_apiclient, mock_moira_client, user_list_data):
+    """
+    Tests that a user can view a video if user is a member of collection's admin_lists
+    """
+    client, _ = logged_in_apiclient
+    user_list_data.collection.admin_lists = [user_list_data.moira_list]
+    user_list_data.collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [user_list_data.moira_list.name]
+    url = reverse('video-detail', kwargs={'video_key': user_list_data.video.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_200_OK
+    assert json.loads(result.context_data['js_settings_json'])['editable'] is True
+
+
+def test_video_detail_no_permission(mock_moira_client, logged_in_apiclient, user_list_data):
+    """
+    Tests that a user cannot view a video if user is not a member of collection's lists
+    """
+    client, _ = logged_in_apiclient
+    user_list_data.collection.view_lists = [user_list_data.moira_list]
+    user_list_data.collection.save()
+    mock_moira_client.return_value.user_lists.return_value = ['other_list']
+    url = reverse('video-detail', kwargs={'video_key': user_list_data.video.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_403_FORBIDDEN
```

Comment:
What do you think of adding fixtures for `admin_user_list_data` and `viewable_user_list_data` (or some similar names)? These would capture the concept of collections that the user has admin or view permissions for. For example, lines 501-503 are just boilerplate to ensure that a user has view permissions for a collection, and it's repeated a few times in other test cases in this file (and quite a lot more in `ui/permissions_test.py`). We can capture that in a fixture:

```python
def viewable_user_list_data(user_list_data, mock_moira_client):
    user_list_data.collection.view_lists = [user_list_data.moira_list]
    user_list_data.collection.save()
    mock_moira_client.return_value.user_lists.return_value = ['other_list']
    return user_list_data
```

This test case could then replace `user_list_data` in the args with `viewable_user_list_data` (same deal for collections that we want the user to have admin permissions for).